AirCasting/SessionViews/ChartViewModel.swift

```diff
@@ -15,79 +15,134 @@ final class ChartViewModel: ObservableObject {
 
     var stream: MeasurementStreamEntity? {
         didSet {
-            guard session.isActive || session.isFollowed || session.status == .NEW else { return }
+            setupDatabaseHook()
             generateEntries()
         }
     }
     @Injected private var persistence: PersistenceController
     @Injected private var settings: UserSettings
 
-    private var timeUnit: TimeInterval {
-        session.isMobile ? .minute : .hour
-    }
-
-    private var mainTimer: Timer?
-    private var firstTimer: Timer?
+    
     private let numberOfEntries = Constants.Chart.numberOfEntries
-
-    private var uiResumedNotificationHandle: Any?
     private var cancellables: [AnyCancellable] = []
 
     deinit {
-        mainTimer?.invalidate()
-        firstTimer?.invalidate()
+        stopTimers()
     }
     
     init(session: SessionEntity) {
         self.session = session
         self.chartStartTime = session.endTime
         self.chartEndTime = session.endTime
-        if session.isActive || session.isFollowed || session.status == .NEW {
-            startTimers(session)
+        if shouldBeUpdatingChartForCurrentSession() {
+            startTimers()
             scheduleUIResumeNotification()
         }
         setupHooks()
     }
+    
+    private func shouldBeUpdatingChartForCurrentSession() -> Bool {
+        session.isActive || session.isFollowed || session.status == .NEW
+    }
 
     private func setupHooks() {
         settings.objectWillChange.sink { [weak self] in
             self?.objectWillChange.send()
             self?.refreshChart()
         }.store(in: &cancellables)
+        setupDatabaseHook()
+    }
+    
+    // MARK: Database observing
+    
+    private var databaseObserver: ChartDatabaseObserver?
+    
+    private func setupDatabaseHook() {
+        databaseObserver = nil
+        guard shouldBeUpdatingChartForCurrentSession(), let sensor = stream?.sensorName else { return }
+        let filter: ChartDatabaseObserverFilter = session.isFixed ? .hour : .minute(countingFrom: session.startTime?.convertedFromUTCToLocal ?? DateBuilder.getRawDate())
+        databaseObserver = ChartDatabaseObserver(session: session.uuid, sensor: sensor, filtered: filter) { [weak self] in
+            guard let self = self else { return }
+            self.log("Measurements change detected")
+            self.generateEntries()
+        }
+    }
+    
+    // MARK: Timers
+    
+    private var timeUnit: TimeInterval {
```

Comment:
The timeUnit is relevant in both implementation (we use it in generate entries method)